Document how to create Grafana dashboard (observability)

dockerymick opened the pull request:
https://github.com/open-cluster-management/backlog/issues/8276

dockerymick committed:
added a reference link to new file for grafana dashboard

dockerymick committed:
adding the new file to the master.adoc file

dockerymick committed:
more updates

dockerymick committed:
adding more to the PR

dockerymick committed:
finishing up with adding info

dockerymick committed:
Updating name for the reference link

swopebe commented on observability/design_grafana.adoc:
```diff
@@ -0,0 +1,95 @@
+
+Complete the following steps to set up the Grafana developer instance:
+
+. Run the `setup-grafana-dev.sh` to setup your Grafana instance. When you run the script the following resources are created, `secret/grafana-dev-config`, `deployment.apps/grafana-dev`, `service/grafana-dev`, `ingress.extensions/grafana-dev`:
```
Little punctuation change here?

When you run the script, the following resources are created:

dockerymick replied:
Updated

swopebe commented on observability/design_grafana.adoc:
```diff
@@ -0,0 +1,95 @@
+ingress.extensions/grafana-dev created
+----
+
+. Switch the user to be a Grafana administrator with the `switch-to-grafana-admin.sh` script. You must navigate to Grafana URL, `https://$ACM_URL/grafana-dev/` and log in. Run the following command:
```
Avoid be verbs when we can per style:

Switch the user role to ......

dockerymick replied:
Updated

swopebe commented on observability/design_grafana.adoc:
```diff
@@ -0,0 +1,95 @@
+
+After you set up the Grafana instance, you can design the dashboard. Complete the following steps to refresh the Grafana console and design your dashboard:
+
+. From the Grafana console, create a new dashboard by selecting the *Add* icon from the navigation panel. Select *Create Dashboard*, and then click *Add new panel*.
```
Can get rid of new, usually per style we don't use it if we say "Can create" (Create and *Add* shows that it is new).

dockerymick replied:
done

swopebe commented on observability/design_grafana.adoc:
```diff
@@ -0,0 +1,95 @@
+
+. From the Grafana console, create a new dashboard by selecting the *Add* icon from the navigation panel. Select *Create Dashboard*, and then click *Add new panel*.
+
+. From the _NewDashboard/Edit Panel_ view, navigate to the _Query_ tab.
```
Does it look like this in the UI?

NewDashboard/Edit Panel_

If so, that should prob change, but we can safely: New Dashboard/Edit Panel

(Should not be capital D and P but if it is in the UI, I would say at least just enter the space.)

dockerymick replied:
@songleo can you help confirm if we wrote the correct name for the Panel? (capitalization and naming)

songleo replied:
looks good to me

dockerymick committed:
Update design_grafana.adoc

dockerymick committed:
Update whats_new.adoc

dockerymick committed:
Merge branch '2.2_stage' into 8276

clyang82 commented on observability/design_grafana.adoc:
```diff
@@ -0,0 +1,95 @@
+[#designing-your-grafana-dashboard]
+= Designing your Grafana dashboard
+
+You can design your Grafana dashboard by creating a MultiClusterObservability custom resource (CR) instance. 
```
Suggest:
`You can design your Grafana dashboard by creating a grafana-dev instance.`

dockerymick replied:
Done

clyang82 commented on observability/design_grafana.adoc:
```diff
@@ -0,0 +1,95 @@
+ingress.extensions/grafana-dev created
+----
+
+. Switch the user role to Grafana administrator with the `switch-to-grafana-admin.sh` script. You must navigate to Grafana URL, `https://$ACM_URL/grafana-dev/` and log in. Run the following command:
```
Do we need replace `Run the following command` with `Then run the following command`?

dockerymick replied:
I can make that change

dockerymick replied:
Done

clyang82 commented:
@dockerymick the client needs to clone this repo - https://github.com/open-cluster-management/multicluster-monitoring-operator/ and then navigate to tools folder so that they can use the shell scripts we mentioned in this document. We need to reflect it in your document. Thanks.

dockerymick commented:
@clyang82 Let me know if I am understanding the request, you would like for me to add a task to tell the user to navigate to /multicluster-monitoring-operator/ tools folder? If that is correct, I disagree with that suggestion bc they have the same steps. We would interrupt the reader and pull them away from the docs.

clyang82 commented:
@dockerymick the document just said that user should run the shell script to setup grafana-dev instance. but the document never mentions where to get the shell scripts.

dockerymick commented:
@clyang82 understood, thanks for clarifying

dockerymick committed:
Update design_grafana.adoc

letting the user know that they must clone the repo

dockerymick commented:
@clyang82 Here is what i added: https://github.com/open-cluster-management/rhacm-docs/pull/1791/files#diff-d4063a33d2e3118eb98e5f73420a00925eeaee630dfe610a61f8094f6df09394R9

clyang82 commented on observability/design_grafana.adoc:
```diff
@@ -0,0 +1,95 @@
+[#setting-up-the-grafana-developer-instance]
+== Setting up the Grafana developer instance
+
+First, clone the link:https://github.com/open-cluster-management/multicluster-monitoring-operator/[`open-cluster-management/multicluster-monitoring-operator/`] repository, so that you are able to run the scripts to set up the instance. Complete the following steps to set up the Grafana developer instance:
```
replace `multicluster-monitoring-operator` to `multicluster-observability-operator`. We are going to rename the repo soon.

so that you are able to run the scripts  -> so that you are able to run the scripts in `tools` folder

dockerymick replied:
done

dockerymick committed:
Update design_grafana.adoc

Updated the alt text for the link, but will need to update the link to the repo soon

dockerymick commented:
I have addressed the comments in this PR and received approval. Thank you @songleo @clyang82 @swopebe for the reviews.
